backend/danswer/indexing/indexing_pipeline.py

```diff
@@ -70,21 +72,43 @@ def _indexing_pipeline(
     """Takes different pieces of the indexing pipeline and applies it to a batch of documents
     Note that the documents should already be batched at this point so that it does not inflate the
     memory requirements"""
-    document_ids = [document.id for document in documents]
-
     with Session(get_sqlalchemy_engine()) as db_session:
-        # acquires a lock on the documents so that no other process can modify them
-        prepare_to_modify_documents(db_session=db_session, document_ids=document_ids)
+        document_ids = [document.id for document in documents]
+
+        # Skip indexing docs that don't have a newer updated at
+        # Shortcuts the time-consuming flow on connector index retries
+        db_docs = get_documents_by_ids(
+            document_ids=document_ids,
+            db_session=db_session,
+        )
+        id_update_time_map = {
+            doc.id: doc.doc_updated_at for doc in db_docs if doc.doc_updated_at
+        }
 
-        # create records in the source of truth about these documents
+        updatable_docs = []
+        for doc in documents:
+            if (
+                doc.id in id_update_time_map
+                and doc.doc_updated_at
+                and doc.doc_updated_at <= id_update_time_map[doc.id]
+            ):
+                continue
+            updatable_docs.append(doc)
+        updatable_ids = [doc.id for doc in updatable_docs]
+
+        # Acquires a lock on the documents so that no other process can modify them
+        prepare_to_modify_documents(db_session=db_session, document_ids=updatable_ids)
+
+        # Create records in the source of truth about these documents,
+        # does not include doc_updated_at which is also used to indicate a successful update
```

Comment:
doesn't seem ideal to overload the `updated_at` field to also be the signal for if a document has been successfully inserted into Vespa. It's prob fine for now, but might be good to mark it as a TODO for later just so we don't forget